src/components/CreateEditForm.tsx

```diff
@@ -0,0 +1,265 @@
+    email: string,
+    firstName: string,
+    lastName: string,
+    phone: number
+}
+const UseStyles = makeStyles({
+    create: {
+
+        width: "100% ",
+        padding: "0",
+
+    },
+    create2: {
+        margin: "10px ",
+        width: "100% ",
+        padding: "0",
+
+    },
+    textname: {
+        width: "100%",
+        marginBottom: "50px",
+    },
+    date: {
+        marginTop: "10px",
+    },
+    box: {
+        display: "flex",
+        flexDirection: "column",
+    },
+    btn: {
+        marginTop: "10px",
+        width: "100%",
+        display: "flex"
+    },
+    form: {
+
+        margin: "0",
+        padding: "0",
+    },
+
+
+    invalidFeedback: {
+        color: "red",
+        fontSize: "small",
+        marginTop: "0",
+        textAlign: "center",
+    },
+    invalidFeedbackForSex: {
+        color: "red",
+        fontSize: "small",
+        marginTop: "0",
+    },
+});
+
+const Create = () => {
+    const classes = UseStyles();
+    const { selectedUser } = React.useContext(UserDataContext);
+    const isAddMode = !selectedUser.id;
+    const { dispatch } = useUsersContext()
+    const [isPending, setPending] = React.useState<boolean>(false);
+
+
+    const validationSchema = yup.object().shape({
+        firstName: yup
+            .string()
+            .matches(/^[aA-zZ\s]+$/, "only alphabets | no special characters")
+            .max(25, "First Name could not be 25 characters long")
+            .min(4, "First Name must contain at least 4 characters")
+            .required("First Name is required"),
+        lastName: yup
+            .string()
+            .matches(/^[aA-zZ\s]+$/, "only alphabets | no special characters")
+            .max(25, "Last Name could not be 25 characters long")
+            .min(4, "Last Name must contain at least 4 characters")
+            .required("Last Name is required"),
+        email: yup.string().email("valid Email is Required").required("valid Email is Required"),
+        phone: yup.number().positive().integer().required("Number is Required"),
+
+    });
+
+    const { register, handleSubmit, reset, setValue, getValues, formState } =
+        useForm<UserType>({
+            resolver: yupResolver(validationSchema),
+        });
+    const { errors } = formState;
+    function onSubmit(data: UserType) {
+        return isAddMode ? createUser(data) : updateUser(selectedUser.id, data);
+    }
+    const update = async () => {
+        try {
+            const resp = await apiFetch("http://localhost:8000/users");
+            dispatch({ type: 'SET_USERS', payload: resp });
+        } catch (err) {
+            console.error(err);
+        }
+    };
+    const updateUser = async (
+        id: number | string,
+        userData: UserType
+    ) => {
+        try {
+            await apiFetch(`http://localhost:8000/users/` + id, {
+                method: "PUT",
+                headers: {
+                    "Content-Type": "application/json",
```

Comment:
We don't need to add headers `"Content-Type": "application/json"` separately everywhere as we are setting this header value in the apiFetch function.